### PR TITLE
Upgrade to 8.1

### PR DESCRIPTION
**Pull request by saraedum**
see https://github.com/conda-forge/sage-feedstock/issues/10 for the issue tracking the dependencies.

### CODE REVIEW

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender

**Comment by saraedum** [comment]
@isuruf Is CI somehow disabled for this repository?

**Comment by saraedum** [comment]
Currently I get to
```
Error compiling Cython file:
------------------------------------------------------------
...
    void   pari_err_ROOTS0(const char *f)

# Auto-generated declarations. There are taken from the PARI version
# on the system, so they more up-to-date than the above. In case of
# conflicting declarations, auto_paridecl should have priority.
from .auto_paridecl cimport *
^
------------------------------------------------------------

/home/jule/proj/conda/miniconda3/envs/conda-build-sagelib/conda-bld/sagelib_1521489491380/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python2.7/site-packages/cypari2/paridecl.pxd:5174:0: 'cypari2/auto_paridecl.pxd' not found
```

**Comment by jakirkham** [comment]
We updated Cython yesterday. So might want to check to see if Cython version affects the error (and thus whether it is an upstream issue).

**Comment by jakirkham** [comment]
Could you please try pushing again to see if CircleCI comes back now? Tried to fix it.

**Comment by saraedum** [comment]
@jakirkham Thanks for the input. The file is missing on conda and I have it with my local install of cypari2, so I think it's a problem with the conda package.

**Comment by saraedum** [comment]
@jakirkham No, CircleCI still does not do anything unfortunately.

**Comment by jakirkham** [comment]
Ah ok. Missed your xref. Sounds reasonable. Would just keep an eye on whether Cython version itself has any effect if the more obvious problems are ruled out.

Thanks for the info. Hmm. Can you please check if `CircleCI` shows up in your `Authorized OAuth Apps` in GitHub's `Settings`?

**Comment by saraedum** [comment]
Yes, CircleCI is there. I also use it all the time.

**Comment by jakirkham** [comment]
Could you please try revoking it and readding it? We have in a few cases seen CircleCI gets a little confused for whatever reason and needs a reset to work on other orgs like conda-forge.

**Comment by saraedum** [comment]
@jakirkham I just tried and it does not seem to work. It builds the branch https://circleci.com/gh/saraedum/sagelib-feedstock/1 but it does not show up here. (I guess it's not working in the conda-forge namespace or however that is called in CircleCI parlance.)

**Comment by saraedum** [comment]
So, it built once for "Upgrade sage's dependencies", the second commit on this PR.

**Comment by jakirkham** [comment]
Ah ok. Thanks for the info.

CircleCI doesn't normally like to build on a user's fork and an org at the same time. It's a little strange. Disabling CircleCI on your fork and repushing *should* help.

**Comment by saraedum** [comment]
I only enabled the build for my repo after it did not work just now. But let me try to stop that integration and push again.

**Comment by saraedum** [comment]
Hm…now I also used the linter?

**Comment by jakirkham** [comment]
Yeah that was my thought too. 😕

If you look at the CircleCI settings on GitHub, does it actually show itself as enabled for conda-forge?

**Comment by saraedum** [comment]
Looks good to me…

2018-03-19-231118_swaygrab

**Comment by saraedum** [comment]
Thanks for providing ideas btw. It's bizarre because it works fine for other projects. I should probably contact CircleCI at some point.

**Comment by jakirkham** [comment]
Sorry you are running into this problem.

Yeah that might be best. Out of ideas for now.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,43 +35,43 @@ requirements:
...
+    - libpng >=1.6.29,<1.7
     - linbox 1.4.*
     - m4ri >=20140914
     - m4rie >=20150908
     - mpc 1.0.*
     - mpfi 1.5.*
     - mpfr 3.*
     - ntl 10.3.*
-    - numpy 1.11.*
+    - numpy 1.13.*
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you fix `numpy` pinning in run?

**Comment by saraedum** [comment]
Compilation ran through and then it fails with:
```
+ python -u setup.py install
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 342, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 333, in execute
    noverify=args.no_verify)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 97, in build
    need_source_download=need_source_download, config=config)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1524, in build_tree
    config=config)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1147, in build
    utils.check_call_env(cmd, env=env, cwd=src_dir)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 628, in check_call_env
    return _func_defaulting_env_to_os_environ(subprocess.check_call, *popenargs, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 624, in _func_defaulting_env_to_os_environ
    return func(_args, **kwargs)
  File "/opt/conda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-e', '/feedstock_root/build_artefacts/sagelib_1522572220102/work/sage-8.1/conda_build.sh']' returned non-zero exit status 1.
+ exit 1
Exited with code 1
```

**Comment by isuruf** [comment]
Can you remove the redirection to /dev/null in `python -u setup.py install > /dev/null 2>&1` so that we can see the actual error?

**Comment by saraedum** [comment]
With the output included, we get:
```
byte-compiling /feedstock_root/build_artefacts/sagelib_1522612852223/_b_env_placehold_plac************************************************************************
Traceback (most recent call last):
  File "setup.py", line 943, in <module>
    ext_modules = ext_modules)
  File "/feedstock_root/build_artefacts/sagelib_1522612852223/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/feedstock_root/build_artefacts/sagelib_1522612852223/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/feedstock_root/build_artefacts/sagelib_1522612852223/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 874, in run
    self.install_kernel_spec()
  File "setup.py", line 889, in install_kernel_spec
    from sage.repl.ipython_kernel.install import SageKernelSpec
  File "/feedstock_root/build_artefacts/sagelib_1522612852223/work/sage-8.1/src/sage/repl/ipython_kernel/install.py", line 16, in <module>
    from jupyter_core.paths import ENV_JUPYTER_PATH
ImportError: No module named jupyter_core.paths
```

**Comment by isuruf** [comment]
Looks like `jupyter_core` is a dependency of sagelib. Can you add it to requirements?

**Comment by saraedum** [comment]
Ok. Build finished, now tests fail:
```
+ source /opt/conda/bin/activate /feedstock_root/build_artefacts/sagelib_1522657960158/_t_env
+ /feedstock_root/build_artefacts/sagelib_1522657960158/_t_env/bin/python -s /feedstock_root/build_artefacts/sagelib_1522657960158/test_tmp/run_test.py
import: 'sage'
+ /bin/bash -x -e /feedstock_root/build_artefacts/sagelib_1522657960158/test_tmp/run_test.sh
+ sage -v
Error: You must set either the SAGE_LOCAL or SAGE_SCRIPTS_DIR environment variable to run this
Error setting environment variables by sourcing '/feedstock_root/build_artefacts/sagelib_1522657960158/_t_env/bin/sage-env';
```

**Comment by isuruf** [comment]
Looks like there needs to be a `sage-env-config` file in `$PREFIX/bin`

**Comment by isuruf** [comment]
Can you revert https://github.com/conda-forge/sagelib-feedstock/pull/16/commits/7c6ab27e6c829bb659bd99d05c65b3e87047c88b so that we can see the errors?

**Comment by saraedum** [comment]
Sorry, I don't understand. Which logs are missing?

**Comment by isuruf** [comment]
"The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over). The job has been terminated"

Travis logs can't go over 4 MB.

**Comment by saraedum** [comment]
Interestng that it just failed now. Anyway, I'll revert it.

**Comment by saraedum** [comment]
Oh, I see. So exceeding the log file size limit just means that you can't see the logs. Whether it failed or not does not depend on that. (That's different from GitLab CI that I usually work with.)

**Comment by saraedum** [comment]
@isuruf I don't really understand how the activation scripts work. `recipe/activate/sage-conda-env.py` seems to be deactivated at the moment (it's not being run by `activate.sh`) or am I misunderstanding something here?
I tried to run doctests with `SAGE_SPKG_INST=`mktemp -d` sage -tp src` in `meta.yml`'s test section. This did not work because sage apparently did not pick the variable up. What am I missing here? (I know that `mktemp -d` does not work on OSX but that's a different issue.)

**Comment by saraedum** [comment]
@isuruf Could you have a look at the output of CircleCI/Travis? Does this otherwise look reasonable?

**Comment by isuruf** [comment]
Activation script just sets `SAGE_ROOT` and `SAGE_LOCAL`. I tried to do a more advanced acivation using sage-env, but it didn't work.

> This did not work because sage apparently did not pick the variable up. What am I missing here? 

Are you sure that this works in sage-8.1? Looks like a sage issue to me.

**Comment by isuruf** [comment]
Btw, for the tests you need all of the standard packages in sage-feedstock right?

**Comment by saraedum** [comment]
Sure, I need the dependencies. I just wanted to see whether I could run the tests already like that.

**Comment by saraedum** [comment]
@isuruf Anyway, is this now good to go?

**Pull request merged**